webapp/components/utils/Notifications.js

```diff
@@ -40,4 +40,117 @@ export const notifications = [
       },
     },
   },
+  {
+    read: false,
+    reason: 'filed_report_on_resource',
+    from: {
+      __typename: 'Report',
+      id: 'reportOnUser',
+      filed: [
+        {
+          reasonCategory: 'discrimination_etc',
+          reasonDescription: 'This user is harassing me with bigoted remarks!',
+          reportedResource: {
+            __typename: 'User',
+            id: 'badWomen',
+            slug: 'mrs.-badwomen',
+            name: 'Mrs. Badwomen',
+          },
+        },
+      ],
+    },
+  },
 ]
+
+export const extractNotificationDataOfCurrentUser = (notification, currentUser) => {
+  const from = notification.from
+  let triggerer
+  const id = from.id
+  const createdAt = notification.createdAt
+  const read = notification.read
+  const reason = notification.reason
+  let title
+  let author
+  let user = null
+  let post = null
+  let comment = null
+  let contentExcerpt = null
+  let report = null
+  let reasonExtention = ''
+
+  if (from.__typename === 'Post') {
+    post = from
+    triggerer = post.author
+  } else if (from.__typename === 'Comment') {
+    comment = from
+    triggerer = comment.author
+    post = comment.post
```

Comment:
This can be refactored. Something like this:
```js
const { post } = from.comment || from
const { author: violator } = from
```
etc.

It's just so hard to read.

Reply:
I tried to code it as clear as possible again! @roschaefer 

I will discuss this with @mattwr18 !